pennylane/ops/qubit/hamiltonian.py

```diff
@@ -616,7 +616,16 @@ def compare(self, other):
         if isinstance(other, Hamiltonian):
             self.simplify()
             other.simplify()
-            return self._obs_data() == other._obs_data()  # pylint: disable=protected-access
+
+            for self_data in self._obs_data():
+                is_equal = False
+                for other_data in other._obs_data():
```

Comment:
You can disable the pylint warning by doing the following:
```suggestion
                for other_data in other._obs_data():    # pylint: disable=protected-access
```

I would suggest only disabling pylint warnings when the warning cannot be fixed by changing the code.